sale_order_type/models/account_move.py

```diff
@@ -12,26 +12,25 @@ class AccountMove(models.Model):
         string="Sale Type",
         compute="_compute_sale_type_id",
         store=True,
-        readonly=True,
-        states={"draft": [("readonly", False)]},
+        readonly=False,
+        states={"posted": [("readonly", True)], "cancel": [("readonly", True)]},
     )
 
     @api.depends("partner_id", "company_id")
     def _compute_sale_type_id(self):
         for record in self:
             if not record.partner_id:
```

Comment:
Shouldn't we also check if `record.is_invoice()`, to avoid setting the sale_order_type if the move is not an invoice?

```py
if not record.is_invoice():
    continue
```